docs/auth/add-auth-provider.md

```diff
@@ -0,0 +1,193 @@
+# Adding authentication providers
+
+## Passport
+
+We chose [Passport](http://www.passportjs.org/) as our authentication platform due to its comprehensive set of supported authentication [strategies](http://www.passportjs.org/packages/).
+
+## How to add a new strategy provider
+
+### Installing the dependencies:
+
+```bash
+yarn add passport-provider-a
+yarn add @types/passport-provider-a
+```
+
+### Create implementation
+
+Make a new folder with the name of the provider following the below file structure:
+
+```bash
+plugins/auth-backend/src/providers/providerA
+├── index.ts
+└── provider.ts
+```
+
+**`plugins/auth-backend/src/providers/providerA/provider.ts`** defines the provider class which implements a handler for the chosen framework.
+
+#### Adding an OAuth based provider
+
+If we're adding an `OAuth` based provider we would implement the [OAuthProviderHandlers](#OAuthProviderHandlers) interface.
+
+The provider class takes the provider's configuration as a class parameter. It also imports the `Strategy` from the passport package.
+
+```ts
+import { Strategy as ProviderAStrategy } from 'passport-provider-a';
+
+export class ProviderAAuthProvider implements OAuthProviderHandlers {
+  private readonly providerConfig: AuthProviderConfig;
+  private readonly _strategy: ProviderAStrategy;
+
+  constructor(providerConfig: AuthProviderConfig) {
+    this.providerConfig = providerConfig;
+    this._strategy = new ProviderAStrategy(
+      { ...providerConfig.options },
+      verifyFunction, // See the "Verify Callback" section
+    );
+  }
+
+  async start() {}
+  async handler() {}
+}
+```
+
+#### Adding an non-OAuth based provider
+
+An non-`OAuth` based provider could implement [AuthProviderRouteHandlers](#AuthProviderRouteHandlers) instead.
```

Comment:
We should add a note saying the non-OAuth provider is in an experimental stage? and would depend on the auth provider that you choose to add?